docs/configuration/config-file-reference.md

```diff
@@ -4340,6 +4340,10 @@ ring:
   # CLI flag: -ruler.ring.final-sleep
   [final_sleep: <duration> | default = 0s]
 
+  # Keep instance in the ring on shut down.
+  # CLI flag: -ruler.ring.keep-instance-in-the-ring-on-shutdown
+  [keep_instance_in_the_ring_on_shutdown: <boolean> | default = false]
```

Comment:
Can we use the same config name as the other services? Ex:

```
# Unregister the compactor during shutdown if true.
  # CLI flag: -compactor.ring.unregister-on-shutdown
  [unregister_on_shutdown: <boolean> | default = true]
```

Reply:
Ruler ring returns `BasicLifecyclerConfig` [here](https://github.com/cortexproject/cortex/blob/master/pkg/ruler/ruler_ring.go#L101). `BasicLifecyclerConfig` uses `KeepInstanceInTheRingOnShutdown` as shown [here](https://github.com/cortexproject/cortex/blob/master/pkg/ring/basic_lifecycler.go#L59)

Reply:
Oh :/

Ok.. I guess we failed to have the same config name in the past.. =/ Very weird to have 2 configs that does the same thing but in different component.. but as we already have another one that use `keep_instance_in_the_ring_on_shutdown` i guess im fine.